From 87e89bd112e16503f37d219a525a5b5d470e08f9 Mon Sep 17 00:00:00 2001
From: Christian Lindig <christian.lindig@citrix.com>
Date: Tue, 30 Oct 2018 10:19:06 +0000
Subject: [PATCH] tools/ocaml: make type of Xsraw.sync more precise

The type of Xsraw.sync is made more precise:

from val sync : (Xenbus.Xb.t -> 'a) -> con -> string
to   val sync : (Xenbus.Xb.t -> unit) -> con -> string

The first argument is enforced to return unit rather than a value that
is not used anyway.

[ No functional change. -iwj ]

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>
Acked-by: Ian Jackson <ian.jackson@eu.citrix.com>
diff --git a/tools/ocaml/libs/xs/xsraw.mli b/tools/ocaml/libs/xs/xsraw.mli
index 57e4fb0..374f0f9 100644
--- a/tools/ocaml/libs/xs/xsraw.mli
+++ b/tools/ocaml/libs/xs/xsraw.mli
@@ -33,7 +33,7 @@ val has_watchevents : con -> bool
 val get_watchevent : con -> string * string
 val read_watchevent : con -> string * string
 val sync_recv : Xenbus.Xb.Op.operation -> con -> string
-val sync : (Xenbus.Xb.t -> 'a) -> con -> string
+val sync : (Xenbus.Xb.t -> unit) -> con -> string
 val ack : string -> unit
 val validate_path : string -> unit
 val validate_watch_path : string -> unit
diff --git a/tools/ocaml/xenstored/process.ml b/tools/ocaml/xenstored/process.ml
index 74c69f8..35353b2 100644
--- a/tools/ocaml/xenstored/process.ml
+++ b/tools/ocaml/xenstored/process.ml
@@ -348,7 +348,7 @@ let transaction_replay c t doms cons =
 				false
 			)
 		(fun () ->
-			Connection.end_transaction c tid None
+			ignore @@ Connection.end_transaction c tid None
 		)
 
 let do_watch con t domains cons data =
@@ -366,7 +366,7 @@ let do_unwatch con t domains cons data =
 		| [node; token; ""]   -> node, token
 		| _                   -> raise Invalid_Cmd_Args
 		in
-	Connections.del_watch cons con node token
+	ignore @@ Connections.del_watch cons con node token
 
 let do_transaction_start con t domains cons data =
 	if Transaction.get_id t <> Transaction.none then
